src/deserialize/parse.rs

```diff
@@ -13,6 +13,14 @@ pub(crate) fn parse(
         Some((_, '(')) => read_list(chars)?,
         Some((_, '#')) => tagged_or_set(chars)?,
         Some((_, '{')) => read_map(chars)?,
+        Some((_, ';')) => {
+            let cmt = read_comment(chars)?;
+            match chars.next() {
```

Comment:
Considering read_comment returns `Edn::Empty` and the functions receives a mutable chars, I don't think this match is necessary

Reply:
This is for the case where the only thing existing is a comment (tests starting on line 815) (fails without match). Normally the return from read_comment is ignored in favor of whatever the next call to parse returns. Your comment made me realize that an attempt to deserialize an empty or whitespace only string will fail instead of return Edn::Empty, so it would be surprising for comment to behave differently. For now I will make both return Edn::Empty, but the other option is to have both fail..